src/de/grobox/liberario/activities/MainActivity.java

```diff
@@ -114,7 +114,11 @@ public void onCreate(Bundle savedInstanceState) {
              .withOnAccountHeaderListener(new AccountHeader.OnAccountHeaderListener() {
 	             @Override
 	             public boolean onProfileChanged(View view, IProfile profile, boolean currentProfile) {
-		             if(!currentProfile && profile != null && profile instanceof ProfileDrawerItem) {
+		             if(currentProfile) {
+			             Intent intent = new Intent(getContext(), PickNetworkProviderActivity.class);
+			             ActivityOptionsCompat options = ActivityOptionsCompat.makeScaleUpAnimation(getCurrentFocus(), 0, 0, 0, 0);
+			             ActivityCompat.startActivityForResult(MainActivity.this, intent, CHANGED_NETWORK_PROVIDER, options.toBundle());
+		             } else if(profile != null && profile instanceof ProfileDrawerItem) {
```

Comment:
Looks like you duplicated this code from elsewhere. How about creating a dedicated method for that code and call it from those two places?

Have you tested this? I ask because you seem to return `false` which means that the event was not consumed. Do you know what effect that has? Will this event then picked up by other code? What happens if you return `true` instead?